components/filter/filter.js

```diff
@@ -389,7 +389,7 @@ class Filter extends LocalizeCoreElement(RtlMixin(LitElement)) {
 					<d2l-list-item
 						?hidden="${item.hidden}"
 						key="${item.key}"
-						label="${item.text}"
+						label="${item.text ? item.text : 'placeholder'}"
```

Comment:
Another thing to note, here is the failing test: https://github.com/BrightspaceUI/core/blob/main/components/filter/test/filter.test.js#L913. Only the arrowLeft testcase is failing... if you swap the order, the escape test fails. 

I have no idea how this could happen as we are creating a new fixture each time. There could be some funkiness happening in the tests, but I haven't found an ordering that allows both test cases to pass.

Reply:
Been digging into this - I think those tests failing are actually a coincidence based on timing.  I think it's some earlier tests causing the issue, and it doesn't get triggered until 3 seconds later (during the arrow tests) because of https://github.com/BrightspaceUI/core/blob/main/mixins/labelled-mixin.js#L212.

Ironically, if I ever get the `LabelledMixin` moved to the new subscription controller this wouldn't be an issue because it cancels label checks for nodes that are no longer connected to the DOM.  But until then it's an easy fix, I'll put up a PR.

Reply:
Thanks for investigating Stacey. It's very interesting that the LabelledMixin throws errors for tests already disconnected 🤔 .